bark/world/tests/evaluation/rss_interface_test.cc

```diff
@@ -0,0 +1,143 @@
+#include "bark/world/objects/agent.hpp"
+#include "bark/world/world.hpp"
+
+#include <ad/map/lane/Operation.hpp>
+#include <ad/map/match/AdMapMatching.hpp>
+#include <ad/map/match/Object.hpp>
+#include <ad/map/point/ENUOperation.hpp>
+#include <ad/map/route/FullRoute.hpp>
+#include <ad/map/route/Planning.hpp>
+#include <ad/rss/core/RssCheck.hpp>
+#include <ad/rss/map/RssSceneCreation.hpp>
+#include <ad/rss/situation/Physics.hpp>
+#include <ad/rss/situation/SituationSnapshot.hpp>
+#include <ad/rss/state/ProperResponse.hpp>
+#include <ad/rss/state/RssStateOperation.hpp>
+#include <ad/rss/state/RssStateSnapshot.hpp>
+#include <ad/rss/world/RssDynamics.hpp>
+
+using bark::world::evaluation::RssInterface;
+
+using namespace bark::geometry;
+using namespace bark::commons;
+using namespace bark::models::behavior;
+using namespace bark::models::dynamic;
+using namespace bark::models::execution;
+using namespace bark::world;
+using bark::world::goal_definition::GoalDefinitionPolygon;
+
+TEST(rss_interface, test_maps_compatibility) {
+  // not all map can be loaded into rss
+  RssInterface rss;
+  ASSERT_TRUE(rss.initializeOpenDriveMap(
+      "bark/runtime/tests/data/city_highway_straight_centered.xodr"));
+  ASSERT_TRUE(rss.initializeOpenDriveMap(
+      "bark/runtime/tests/data/DR_DEU_Merging_MT_v01_centered.xodr"));
+  ASSERT_TRUE(rss.initializeOpenDriveMap(
+      "bark/runtime/tests/data/DR_CHN_Merging_ZS_partial_v02.xodr"));
+}
+
+TEST(rss_interface, test_rss_check) {
+  auto params = std::make_shared<SetterParams>(false);
+
+  BehaviorModelPtr beh_model(new BehaviorConstantAcceleration(params));
+  DynamicModelPtr dyn_model(new SingleTrackModel(params));
+  ExecutionModelPtr exec_model(new ExecutionModelInterpolate(params));
+
+  Polygon shape = standard_shapes::CarRectangle();
+  Polygon polygon(
+      Pose(0, 0., 0),
+      std::vector<Point2d>{Point2d(0, 0), Point2d(-1, -1), Point2d(-1, 1),
+                           Point2d(1, 1), Point2d(1, -1)});
+  std::shared_ptr<Polygon> goal_polygon(
+      std::dynamic_pointer_cast<Polygon>(polygon.Translate(Point2d(-16, 108))));
+
+  auto goal_definition_ptr =
+      std::make_shared<GoalDefinitionPolygon>(*goal_polygon);
+
+  State state(static_cast<int>(StateDefinition::MIN_STATE_SIZE));
+  state << 0, 68, 108, 0, 10;
+  objects::AgentPtr agent(new Agent(state, beh_model, dyn_model, exec_model,
+                                    shape, params, goal_definition_ptr));
+
+  WorldPtr world(new World(params));
+  world->AddAgent(agent);
+
+  RssInterface rss(
+      "bark/runtime/tests/data/DR_DEU_Merging_MT_v01_centered.xodr",
+      std::vector<float>{3.5, -8., -4., -3., 0.2, -0.8, 0.1, 1.},
+      std::unordered_map<AgentId, std::vector<float>>(), 1, 50);
+
+  ::ad::rss::world::WorldModel rss_world;
+  ASSERT_TRUE(rss.ExtractRSSWorld(*world, agent->GetAgentId(), rss_world));
+
+  ::ad::rss::state::RssStateSnapshot snapshot;
+  ASSERT_TRUE(rss.RssCheck(rss_world, snapshot));
```

Comment:
this test could be a bit more elaborate

Reply:
Improved in https://github.com/bark-simulator/bark/commit/d351f05b8f6a89515edf541c2f6e457b98a0d8d6